superset/views/core.py

```diff
@@ -2752,12 +2752,10 @@ def queries(self, last_updated_ms):
         ]
 
         if queries_to_timeout:
-            update(Query).where(
-                and_(
-                    Query.user_id == g.user.get_id(),
-                    Query.client_id in queries_to_timeout,
-                ),
-            ).values(state=QueryStatus.TIMED_OUT)
+            for q in sql_queries:
```

Comment:
How many SQL queries are there? It may be more efficient to have the database handle this logic.